plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,655 @@
+                                .map(IgniteTableProperties.IgniteWriteSyncMode::valueOf)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case CACHE_NAME_PROPERTY:
+                    case CACHE_GROUP_PROPERTY:
+                    case DATA_REGION_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + property);
+                }
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.build();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
+    private List<String> extractColumnNamesFromOrderingScheme(String row)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        if (isNullOrEmpty(row)) {
+            return builder.build();
+        }
+        for (String key : SPLITTER.split(row)) {
+            int left = key.indexOf("\"") + 1;
+            int right = key.lastIndexOf("\"");
+            builder.add(key.substring(left, right).toLowerCase(ENGLISH));
+        }
+        return builder.build();
+    }
+
+    private static SliceWriteFunction uuidWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long high = reverseBytes(value.getLong(0));
+            long low = reverseBytes(value.getLong(SIZE_OF_LONG));
+            UUID uuid = new UUID(high, low);
+            statement.setObject(index, uuid, Types.OTHER);
+        };
+    }
+
+    private static Slice uuidSlice(UUID uuid)
+    {
+        return wrappedLongArray(
+                reverseBytes(uuid.getMostSignificantBits()),
+                reverseBytes(uuid.getLeastSignificantBits()));
+    }
+
+    private ColumnMapping uuidColumnMapping()
+    {
+        return ColumnMapping.sliceMapping(
+                uuidType,
+                (resultSet, columnIndex) -> uuidSlice((UUID) resultSet.getObject(columnIndex)),
+                uuidWriteFunction());
+    }
+
+    public static SliceWriteFunction longDecimalWriteFunction(int precision, int scale)
+    {
+        String bindExpression = format("CAST(? AS decimal(%s, %s))", precision, scale);
+        return new SliceWriteFunction()
+        {
+            @Override
+            public String getBindExpression()
+            {
+                return bindExpression;
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, Slice value)
+                    throws SQLException
+            {
+                statement.setBigDecimal(index, new BigDecimal(value.getLong(0)));
```

Comment:
Shouldn't this be something like? LongDecimalType is packed into a long and needs unpacking.

```
BigInteger unscaledValue = decodeUnscaledValue(value);
BigDecimal bigDecimal = new BigDecimal(unscaledValue, decimalType.getScale(), new MathContext(decimalType.getPrecision()));
statement.setBigDecimal(index, bigDecimal);
```